src/Command/PullRequest/PullRequestMergeCommand.php

```diff
@@ -141,6 +147,12 @@ protected function execute(InputInterface $input, OutputInterface $output)
             $targetLabel = sprintf('Target: %s/%s', $targetRemote, $targetBranch);
         }
 
+        $authenticatedUser = $this->getParameter($input, 'authentication')['username'];
+        $removeSourceBranch = $input->getOption('remove-source-branch');
+        if ($removeSourceBranch && $pr['user'] !== $authenticatedUser) {
```

Comment:
@sstok, I don't know if I got your point. I think this restriction isn't driven by permissions, but for respect to PR author. By instance, in my private organization I have access to delete branches from another author than me, but I think this isn't a good practice or something we should allow. IMO, the responsibility over each branch belongs only to their author, regardless your access level as merger.